javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/InstrumentationModuleClassLoader.java

```diff
@@ -0,0 +1,252 @@
+    if (cachedLookup == null) {
+      // Load the injected copy of LookupExposer and invoke it
+      try {
+        // we don't mind the race condition causing the initialization to run multiple times here
+        Class<?> lookupExposer = loadClass(LookupExposer.class.getName());
+        cachedLookup = (MethodHandles.Lookup) lookupExposer.getMethod("getLookup").invoke(null);
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
+      }
+    }
+    return cachedLookup;
+  }
+
+  @Override
+  protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+    synchronized (getClassLoadingLock(name)) {
+      Class<?> result = findLoadedClass(name);
+
+      // This CL is self-first: Injected class are loaded BEFORE a parent lookup
+      if (result == null) {
+        ClassCopySource injected = getInjectedClass(name);
+        if (injected != null) {
+          byte[] bytecode = injected.getBytecode();
+          if (System.getSecurityManager() == null) {
+            result = defineClassWithPackage(name, bytecode);
+          } else {
+            result =
+                AccessController.doPrivileged(
+                    (PrivilegedAction<Class<?>>) () -> defineClassWithPackage(name, bytecode));
+          }
+        }
+      }
+      if (result == null) {
+        result = tryLoad(agentOrExtensionCl, name);
+      }
+      if (result == null) {
+        result = tryLoad(instrumentedCl, name);
+      }
+
+      if (result != null) {
+        if (resolve) {
+          resolveClass(result);
+        }
+        return result;
+      } else {
+        throw new ClassNotFoundException(name);
+      }
+    }
+  }
+
+  private static Class<?> tryLoad(ClassLoader cl, String name) {
+    try {
+      return cl.loadClass(name);
+    } catch (ClassNotFoundException e) {
+      return null;
+    }
+  }
+
+  @Override
+  public URL getResource(String resourceName) {
+    String className = resourceToClassName(resourceName);
+    if (className != null) {
+      // for classes use the same precedence as in loadClass
+      ClassCopySource injected = getInjectedClass(className);
+      if (injected != null) {
+        return injected.getUrl();
+      }
+      URL fromAgentCl = agentOrExtensionCl.getResource(resourceName);
+      if (fromAgentCl != null) {
+        return fromAgentCl;
+      }
+      return instrumentedCl.getResource(resourceName);
+    } else {
+      // delegate to just the default parent (the agent classloader)
+      return super.getResource(resourceName);
+    }
+  }
+
+  @Override
+  public Enumeration<URL> getResources(String resourceName) throws IOException {
+    String className = resourceToClassName(resourceName);
+    if (className != null) {
+      URL resource = getResource(resourceName);
+      List<URL> result =
+          resource != null ? Collections.singletonList(resource) : Collections.emptyList();
+      return new EnumeratorFromIterator<>(result.iterator());
```

Comment:
Perhaps could use `Collections.enumeration(result)` instead of `EnumeratorFromIterator`

Reply:
Totally missed that that method exists, agree 100%.